### PR TITLE
feat(Gmail Trigger Node): Add filter option to include drafts

### PR DESCRIPTION
**Pull request by riascho**
## Summary

The Gmail Trigger Node was triggering on `draft` messages as well (e.g. when a new draft was created or a draft deleted). 
This PR introduces a new filter to include these `draft` message events in the trigger poll. 

By default the `includeDrafts` filter will be false. 

## Related Linear tickets, Github issues, and Community forum posts

- https://github.com/n8n-io/n8n/issues/9204
- https://linear.app/n8n/issue/NODE-1315/gmail-trigger-ignore-drafts

## Review / Merge checklist

- [x] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) 
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included. 
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

### CODE REVIEW

**Comment by Joffcom** [comment]
We should probably not change the default behaviour as users may have workarounds in place so this could break workflows. Instead for v1 set it to true to include the drafts and make a light vesion for 1.1 and set that to false as the default.

File: packages/nodes-base/nodes/Google/Gmail/GmailTrigger.node.ts [hunk 1]
```diff
@@ -292,8 +307,14 @@ export class GmailTrigger implements INodeType {
 					{},
 					qs,
 				);
-
-				if (!simple) {
+				if (!includeDrafts) {
+					if (responseData[i].labelIds.includes('DRAFT')) {
+						responseData.splice(i, 1);
+						i--;
```
Comment (ShireenMissi) [inline packages/nodes-base/nodes/Google/Gmail/GmailTrigger.node.ts:313]:
```suggestion
						
```

Comment (riascho) [reply packages/nodes-base/nodes/Google/Gmail/GmailTrigger.node.ts:313]: My reasoning was that the splicing shifts the next element to the current index and because we wanna check that element too we would have to reset the index.

Comment (ShireenMissi) [reply packages/nodes-base/nodes/Google/Gmail/GmailTrigger.node.ts:313]: oh I see, although this solution works we should avoid modifying the array as we iterate over it, it also make the code hard to follow.
 You can define another array and add filtered results to it to keep responseData intact while iterating

Comment (riascho) [reply packages/nodes-base/nodes/Google/Gmail/GmailTrigger.node.ts:313]: Ok got it :)  
I had considered using `.filter()` before but thought that `.splice()` is better in terms of memory.
Will update to use `.filter()` instead 👍

**Comment by riascho** [comment]
Hi @ShireenMissi - I ended up not using `.filter()` because I wanted to filter out draft emails while still in the `for` loop. Now I'm just using a new array `withoutDrafts` that stores all messages that are not drafts and we skip parsing if it is a draft email. That way we can avoid some unnecessary data processing. Let me know if this approach is better
Thank You 😊

File: packages/nodes-base/nodes/Google/Gmail/GmailTrigger.node.ts [hunk 2]
```diff
@@ -284,6 +291,15 @@ export class GmailTrigger implements INodeType {
 				qs.format = 'raw';
 			}
 
+			let includeDrafts;
+			if (node.typeVersion > 1.1) {
+				includeDrafts = (qs.includeDrafts as boolean) || false;
```
Comment (ShireenMissi) [inline packages/nodes-base/nodes/Google/Gmail/GmailTrigger.node.ts:296]:
```suggestion
				includeDrafts = (qs.includeDrafts as boolean) ?? false;
```

File: packages/nodes-base/nodes/Google/Gmail/GmailTrigger.node.ts [hunk 3]
```diff
@@ -284,6 +291,15 @@ export class GmailTrigger implements INodeType {
...
+			} else {
+				includeDrafts = (qs.includeDrafts as boolean) || true;
```
Comment (ShireenMissi) [inline packages/nodes-base/nodes/Google/Gmail/GmailTrigger.node.ts:298]: having || true means it will always be true, you can test it by using a node with version older than 1.2 and have include drafts object set to false, with the current setup that doesn't work
```suggestion
				includeDrafts = (qs.includeDrafts as boolean) ?? true;
```

**Comment by ShireenMissi** [comment]
> Hi @ShireenMissi - I ended up not using `.filter()` because I wanted to filter out draft emails while still in the `for` loop. Now I'm just using a new array `withoutDrafts` that stores all messages that are not drafts and we skip parsing if it is a draft email. That way we can avoid some unnecessary data processing. Let me know if this approach is better Thank You 😊

Thanks for making the changes @riascho 🙏
one final comment and it will be ready to go

File: packages/nodes-base/nodes/Google/Gmail/GmailTrigger.node.ts (hunk 3 above)
Comment (riascho) [reply packages/nodes-base/nodes/Google/Gmail/GmailTrigger.node.ts:298]: Thanks Shireen! Makes a lot of sense :) Nice refresher for me on Nullish Coalescing...

**Review by ShireenMissi — APPROVED** [review]
Tested manually and it works 🎉

**Comment by janober** [comment]
Got released with [`n8n@1.67.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.67.0)
